src/metadata.rs

```diff
@@ -501,38 +573,54 @@ where
                 break;
             }
         }
-
-        if signatures_needed == 0 {
-            Ok(())
-        } else {
-            Err(Error::VerificationFailure(format!(
+        if signatures_needed > 0 {
+            return Err(Error::VerificationFailure(format!(
                 "Signature threshold not met: {}/{}",
                 threshold - signatures_needed,
                 threshold
-            )))
+            )));
         }
+
+        // "assume" the metadata is valid because we just verified that it is.
+        self.assume_valid()
     }
 }
 
-impl<D, M> AsRef<M> for SignedMetadata<D, M> {
-    fn as_ref(&self) -> &M {
-        &self.metadata
+impl<D> SignedMetadata<D, RootMetadata>
+where
+    D: DataInterchange,
+{
+    /// Parse common metadata fields from this metadata without verifying signatures.
+    ///
+    /// This operation is generally unsafe to do with metadata obtained from an untrusted source,
+    /// but rolling forward to the most recent root.json requires using the version number of the
+    /// latest root.json.
+    pub fn untrusted_info(&self) -> Result<MetadataMetadata<RootMetadata>> {
```

Comment:
I don't think this needs to be public, you didn't use it in any of the tests.

Reply:
fixed.